### PR TITLE
Edge automation and capability sync

### PR DESCRIPTION
**Pull request by kristofvancoillie**
### Description of your changes

Added functionality to automate the edge interactions (upload files, use `edgecli`) and synchronize the capability.

### JIRA reference

N/A

### Impact Analysis

More functionality which allows for more automation

#### Checklist

- [x] I have performed a self-review of my code
- [x] My code follows the contribution guidelines of this project
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/edge.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                username=self.username,
+                key_filename=self.certificate,
+                timeout=20,
+                look_for_keys=False,
+            )
+            logging.info(f"Connected to {self.address} over SSH")
+
+        except Exception as e:
+            ssh_client = None
+            logging.error(f"Unable to connect to {self.address} over SSH: {e}")
```
Comment (jeromartin) [inline src/edge.py:33]: Isn't it easier to raise exception here and later in the code to assume that `ssh_client` is not `Optional`? I'm not sure about the value of letting tool continue if this step fails

File: src/edge.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                        prevdata = stdout.channel.recv(1024)
+                        alldata += prevdata
+                    logging.info(str(alldata))
+        else:
+            logging.error(f"Connection to Edge server {self.address} not opened.")
+
+    def upload_folder(self, source_folder: str, target_folder: str) -> None:
+        if self.ssh_client:
+            scp = SCPClient(self.ssh_client.get_transport())
+            scp.put(files=source_folder, remote_path=target_folder, recursive=True)
```
Comment (jeromartin) [inline src/edge.py:54]: Do we know what happens if existing files are already there ?

Comment (kristofvancoillie) [reply src/edge.py:54]: it overwrites:
```
[kristof.vancoillie@pm-lineage-demo-edge cl3]$ ls -lah
total 16K
drwxrwxr-x. 2 kristof.vancoillie kristof.vancoillie   66 Jul 23 14:11 .
drwxrwxrwt. 4 root               root                 96 Jul 25 12:00 ..
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie 6.1K Jul 25 14:14 .DS_Store
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie 2.6K Jul 25 14:14 lineage_1.json
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie  551 Jul 25 14:14 metadata.json
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
[kristof.vancoillie@pm-lineage-demo-edge cl3]$ ls -lah
total 16K
drwxrwxr-x. 2 kristof.vancoillie kristof.vancoillie   66 Jul 23 14:11 .
drwxrwxrwt. 4 root               root                 96 Jul 25 12:00 ..
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie 6.1K Jul 25 14:16 .DS_Store
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie 2.6K Jul 25 14:16 lineage_1.json
-rw-r--r--. 1 kristof.vancoillie kristof.vancoillie  551 Jul 25 14:16 metadata.json
[kristof.vancoillie@pm-lineage-demo-edge cl3]$
```

**Review by jeromartin — APPROVED** [review]
👍

**Pull request merged**